pkg/plugins/middlewarewasm.go

```diff
@@ -135,7 +136,15 @@ func (b *wasmMiddlewareBuilder) buildMiddleware(ctx context.Context, next http.H
 		return nil, nil, fmt.Errorf("creating middleware: %w", err)
 	}
 
-	return mw.NewHandler(ctx, next), applyCtx, nil
+	h := mw.NewHandler(ctx, next)
+	runtime.SetFinalizer(h, func(_ http.Handler) {
```

Comment:
please add the following explanation

https://github.com/traefik/traefik/pull/11150/files#diff-b5f8b248fe73ea05a1a712c7863a3668407004105f1cbeba310bdb4675d7d3a4R139-R141